compiler/codegen/cg-expr.cpp

```diff
@@ -6299,6 +6299,44 @@ DEFINE_PRIM(BREAKPOINT) {
   }
 }
 
+DEFINE_PRIM(CONST_ARG_HASH) {
+  INT_ASSERT(call->numActuals() == 1);
+  INT_ASSERT(call->get(1)->isRefOrWideRef());
+  GenRet arg = call->get(1);
+
+  if (call->get(1)->isWideRef()) {
+    Symbol* sym = call->get(1)->typeInfo()->getField("addr", true);
+
+    // Don't send arg in with `&`
+    GenRet ptr = codegenValue(arg);
+    // Do shallow GET on record
+    GenRet remotePtr = codegenValue(codegenDeref(ptr));
+    // Create temporary that we can pass a pointer to
```

Comment:
I recommend you use `createTempVarWith(remotePtr)`. Also since `remotePtr` stores the value, maybe we should rename it.

```
    // Don't send arg in with `&`
    GenRet ptr = codegenValue(arg);
    // Do shallow GET on record
    GenRet val = codegenValue(codegenDeref(ptr));
    // Create temporary that we can pass a pointer to
    GenRet tmp = createTempVarWith(val);
    GenRet addr = codegenAddrOf(tmp);
```